inference-engine/tests/functional/inference_engine/snippets/memory_ops.cpp

```diff
@@ -0,0 +1,99 @@
+        auto data = std::make_shared<opset1::Parameter>(element::f32, Shape{2, 2});
+        auto neg = std::make_shared<opset1::Negative>(data);
+        f = std::make_shared<Function>(NodeVector{neg}, ParameterVector{data});
+
+        pass::Manager m;
+        m.register_pass<pass::InitNodeInfo>();
+        m.register_pass<snippets::pass::InsertLoad>();
+        m.register_pass<snippets::pass::InsertStore>();
+        m.run_passes(f);
+        ASSERT_NO_THROW(check_rt_info(f));
+    }
+    {
+        auto data = std::make_shared<opset1::Parameter>(element::f32, Shape{2, 2});
+        auto load = std::make_shared<snippets::isa::Load>(data);
+        auto neg = std::make_shared<opset1::Negative>(load);
+        auto store = std::make_shared<snippets::isa::Store>(neg);
+        f_ref = std::make_shared<Function>(NodeVector{store}, ParameterVector{data});
+    }
+
+    auto res = compare_functions(f, f_ref);
+    ASSERT_TRUE(res.first) << res.second;
+}
+
+TEST(TransformationTests, InsertLoadTwise) {
+    std::shared_ptr<Function> f(nullptr), f_ref(nullptr);
+    {
+        auto data = std::make_shared<opset1::Parameter>(element::f32, Shape{2, 2});
+        auto neg = std::make_shared<opset1::Negative>(data);
+        f = std::make_shared<Function>(NodeVector{neg}, ParameterVector{data});
+
+        pass::Manager m;
+        m.register_pass<pass::InitNodeInfo>();
+        m.register_pass<snippets::pass::InsertLoad>();
+        m.register_pass<snippets::pass::InsertLoad>();
+        m.run_passes(f);
+        ASSERT_NO_THROW(check_rt_info(f));
+    }
+    {
+        auto data = std::make_shared<opset1::Parameter>(element::f32, Shape{2, 2});
+        auto load = std::make_shared<snippets::isa::Load>(data);
+        auto neg = std::make_shared<opset1::Negative>(load);
+        f_ref = std::make_shared<Function>(NodeVector{neg}, ParameterVector{data});
+    }
+
+    auto res = compare_functions(f, f_ref);
+    ASSERT_TRUE(res.first) << res.second;
+}
+
+TEST(TransformationTests, InsertStoreTwise) {
+    std::shared_ptr<Function> f(nullptr), f_ref(nullptr);
+    {
+        auto data = std::make_shared<opset1::Parameter>(element::f32, Shape{2, 2});
+        auto neg = std::make_shared<opset1::Negative>(data);
+        f = std::make_shared<Function>(NodeVector{neg}, ParameterVector{data});
+
+        pass::Manager m;
+        m.register_pass<pass::InitNodeInfo>();
+        m.register_pass<snippets::pass::InsertStore>();
+        m.register_pass<snippets::pass::InsertStore>();
+        m.run_passes(f);
+        ASSERT_NO_THROW(check_rt_info(f));
+    }
+    {
+        auto data = std::make_shared<opset1::Parameter>(element::f32, Shape{2, 2});
+        auto neg = std::make_shared<opset1::Negative>(data);
+        auto store = std::make_shared<snippets::isa::Store>(neg);
+        f_ref = std::make_shared<Function>(NodeVector{store}, ParameterVector{data});
+    }
+
+    auto res = compare_functions(f, f_ref);
+    ASSERT_TRUE(res.first) << res.second;
+}
```

Comment:
looks like we need more tests :)

Reply:
Yes, we are 👍   they are in progress right now 🤣

Reply:
✅